packages/auth/ios/RNFBAuth/RNFBAuthModule.m

```diff
@@ -939,6 +941,181 @@ - (void)invalidate {
       [[FIRAuth authWithApp:firebaseApp] useEmulatorWithHost: host port:port];
 }
 
+RCT_EXPORT_METHOD(multiFactorEnrollWithPhone:
+  (FIRApp *) firebaseApp
+    :(NSString *) phoneNumber
+    :(RCTPromiseResolveBlock) resolve
+    :(RCTPromiseRejectBlock) reject
+) {
+    FIRUser *user = [FIRAuth authWithApp:firebaseApp].currentUser;
+
+    if (user) {
```

Comment:
These are complicated enough logic blocks, I think both would benefit from un-nesting via a "fast error" path on no user. What I mean specifically:

```suggestion
    if (!user) {
       [self promiseNoUser:resolve rejecter:reject isError:YES];
       return;
    }
```

... now the entire rest of the method may be de-indented and we may proceed with confidence knowing we have a user